src/snowflake/cli/api/commands/flags.py

```diff
@@ -38,8 +42,18 @@
 
 def _connection_callback(prop: str):
     """Generates a setter for a field on the current context manager's connection context."""
+    if not prop in ConnectionContext.__dataclass_fields__:
```

Comment:
this is probably fine, but the public API for this is `fields(ConnectionContext)`

Reply:
It's slightly different -- fields gives you a list of tuples. This is a dict, which is better from a performance perspective for us.